lib/swagger-scanner.ts

```diff
@@ -170,4 +193,71 @@ export class SwaggerScanner {
     );
     return modulePath ?? Reflect.getMetadata(MODULE_PATH, metatype);
   }
+
+  private scanModuleImportsRecursively(
+    imports: Set<Module>,
+    container: NestContainer,
+    currentDepth: number,
+    maxDepth: number | undefined,
+    processedModules: Set<string>,
+    options: {
+      globalPrefix: string;
+      internalConfigRef: ApplicationConfig;
+      operationIdFactory?: OperationIdFactory;
+      linkNameFactory?: (
+        controllerKey: string,
+        methodKey: string,
+        fieldKey: string
+      ) => string;
+      autoTagControllers?: boolean;
+    }
+  ): ModuleRoute[] {
+    let result: ModuleRoute[] = [];
+
+    for (const { metatype, controllers, imports: subImports } of imports.values()) {
+      // Skip if module has already been processed
+      const moduleId = this.getModuleId(metatype);
+      if (processedModules.has(moduleId) || container.isGlobalModule(metatype) || (maxDepth !== undefined && currentDepth > maxDepth)) {
+        continue;
+      }
+      processedModules.add(moduleId);
+
+      // Scan current module's controllers
+      const modulePath = this.getModulePathMetadata(container, metatype);
+      result = result.concat(
+        this.scanModuleControllers(
+          controllers,
+          modulePath,
+          options.globalPrefix,
+          options.internalConfigRef,
+          options.operationIdFactory,
+          options.linkNameFactory,
+          options.autoTagControllers
+        )
+      );
+
+      // Process sub-imports if any
+      if (subImports.size > 0) {
+        const nextDepth = currentDepth + 1;
+        if (maxDepth === undefined || nextDepth < maxDepth) {
+          result = result.concat(
+            this.scanModuleImportsRecursively(
+              subImports,
+              container,
+              nextDepth,
+              maxDepth,
+              processedModules,
+              options
+            )
+          );
+        }
+      }
+    }
+
+    return result;
+  }
+
+  private getModuleId(metatype: Type<any>): string {
+    return metatype.name || Math.random().toString(36).substring(2);
```

Comment:
Not sure if I understand what's the purpose of having `Math.random().toString(36).substring(2)` if it isn't calculated based on the input argument value (metatype)

Reply:
You're right about the random fallback being unnecessary. Let me explain the context better. The `getModuleId` is used with `processedModules` set to prevent duplicate scanning and circular dependencies while recursively scanning module imports. However, I realized that using just `metatype.name` could be problematic - if we have two different modules with the same name in different locations (like auth.module.ts in different directories), only one would be scanned. To properly handle such cases while still preventing infinite recursion, I think we should use `metatype.toString()`. While this generates longer strings, it ensures we don't miss any modules during the scanning process. What do you think about this approach?
```suggestion
    return metatype.toString();
```

Reply:
Sounds reasonable. I'm afraid the size of the array might become notably large for bigger projects though

Reply:
Looking at `ModuleTokenFactory.create`, I see that `id` uses uid(21) and `token` adds metatype.name and dynamicModuleMetadata(optional) to that. Since `NestContainer.addModule` seems to use `token` as the identifier, rather than using metatype, what if we destructure `token` along with the existing properties in the `imports.values()` loop within the `scanModuleImportsRecursively` method and use that as the moduleId?